sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala

```diff
@@ -850,7 +851,14 @@ https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C
         case Token("TOK_TABLESPLITSAMPLE",
                Token("TOK_PERCENT", Nil) ::
                Token(fraction, Nil) :: Nil) =>
-          Sample(fraction.toDouble, withReplacement = false, (math.random * 1000).toInt, relation)
+          // RDD's sample function is on interval [0, 1], but HiveQL's tablesample is on
+          // interval [0, 100]. So we need to judge the interval individually.
```

Comment:
How about "The range of fraction accepted by Sample is [0, 1]. Because Hive's block sampling function takes X PERCENT as the input and the range of X is [0, 100], we need to adjust the fraction."?